node_tree.go

```diff
@@ -258,15 +267,32 @@ func (t *NodeTree) Get(url string) (ok bool, n *Node, err error) {
 	return false, &Node{}, nil
 }
 
-// FullTextSearch uses a prebuild search index to perform a search
+// FullTextSearch uses a prebuilt search index to perform a search
 // over all possible attributes of each node.
 func (t *NodeTree) FullTextSearch(query string) ([]*Node, int, time.Duration) {
 	start := time.Now()
 
-	t.RLock()
-	defer t.RUnlock()
+	textQuery := bleve.NewMatchQuery(query)
+	textQuery.SetFuzziness(2)
+	prefixQuery := bleve.NewPrefixQuery(query)
+	termQuery := bleve.NewTermQuery(query)
+	disjunctionQuery := bleve.NewDisjunctionQuery(prefixQuery, textQuery, termQuery)
+
+	bSearch := bleve.NewSearchRequest(disjunctionQuery)
```

Comment:
I think it'd be nice to get rid of one of the temporary variables above, so I'd suggest inlining what is assigned to `disjunctionQuery` here.

Reply:
Just a little bit of Python creeping in here. :wink: